### ISSUE TITLE
Rule suggestion: MSTest framework silently ignores `async void` test methods

### ISSUE THREAD

**Issue by shtratos** [dotnet/roslyn-analyzers#6042]
### Describe the problem you are trying to solve

MSTest framework silently ignores `async void` test methods.
When regular tests are converted to async ones it's very easy to forget to change the signature of test method.
This is hard to catch during code reviews as the test code is still there, it just does not run despite having `[TestMethod]` annotation.

### Describe suggestions on how to achieve the rule

**How to detect?**

- Look at the classes, annotated with `[TestClass]`
- for each async method annotated with `[TestMethod]`
  - check if return type is `void`?
  - if yes, emit a warning that the test method is going to be silently ignored despite the `[TestMethod]` annotation.
 
**Code fix**: Replace `async void` with `async Task`

### Additional context

Example

```csharp
using System;
using System.Threading.Tasks;

using Microsoft.VisualStudio.TestTools.UnitTesting;

[TestClass]
public class SampleTest
{
    [TestMethod]
    public async void This_Test_Is_Ignored() // analyser SHOULD fire here
    {
        await Task.Run(() => throw new NotSupportedException("BLAM!"));
    }

    [TestMethod]
    public async Task This_Test_Is_Detected() // the analyser SHOULD NOT fire here
    {
        await Task.Run(() => throw new NotSupportedException("BLAM!"));
    }
}

```

**Comment by shtratos** [comment]
My concern is mostly about MSTest here, so that analyzer should probably only be shipped with MSTest package.
Not sure about other testing frameworks and whether they have a similar problem.

**Comment by Evangelink** [comment]
Closing here and following on mstest: https://github.com/microsoft/testfx/issues/1141
